docs/user-manual/plugin-system.md

```diff
@@ -0,0 +1,255 @@
+
+```cpp
+// Shape.h
+class Shape {
+  virtual ~Shape() {}
+  virtual void draw() = 0;
+};
+```
+
+This is our interface definition. A simple **Shape** class with an abstract draw() method. This file is part of our library.
+
+```cpp
+// Square.cc 
+#include <dqm4hep/Logging.h>
+#include <dqm4hep/PluginManager.h>
+#include <Shape.h>
+
+class Square : public Shape {
+  void draw();
+};
+
+void Square::draw() {
+  dqm_info( "Drawing a square !!" );
+}
+
+DQM_PLUGIN_DECL( Square , "SquareShapePlugin" );
+```
+
+```cpp
+// Circle.cc 
+#include <dqm4hep/Logging.h>
+#include <dqm4hep/PluginManager.h>
+#include <Shape.h>
+
+class Circle : public Shape {
+  void draw();
+};
+
+void Circle::draw() {
+  dqm_info( "Drawing a circle !!" );
+}
+
+DQM_PLUGIN_DECL( Circle , "CircleShapePlugin" );
+```
+
+These two files implement our **Shape** interface and just make a print out. They are both compiled in the same library called *libShapePlugins.so*
+
+<div class="info-msg">
+  <i class="fa fa-info-circle"></i>
+  Note that it is not needed to define a header file to implement a plugin !
+</div>
+
+Now, we define our main in this way : 
+
+```cpp
+#include <dqm4hep/PluginManager.h>
+#include <Shape.h>
+
+using namespace dqm4hep::core;
+
+int main(int argc, char** argv) {
+  
+  std::string pluginName( argv[1] );
+  
+  auto mgr = PluginManager::instance();
+  THROW_RESULT_IF( STATUS_CODE_SUCCESS, != , mgr->loadLibraries() );
+
+  Shape* shape = mgr->create<Shape>( pluginName );
+  shape->draw();
+  delete shape;
+
+  return 0;
+}
+```
+
+Compile it as **shape_example** and run it like this : 
+
+```shell
+export DQM4HEP_PLUGIN_DLL=libShapePlugins.so
+./shape_example CircleShapePlugin # to draw a circle
+./shape_example SquareShapePlugin # to draw a square
+```
+
+Note that in this example, your executable is not aware of the existence of the two plugin implementations at compile time. The classes are loaded and hooked via the plugin manager **at runtime**.
+
+# Known plugin interfaces in the framework
+
+A set of interfaces have been pre-defined in DQM4hep and can be implemented as plugins:
+
+- **dqm4hep::core::QualityTest** (*dqm4hep-core* package)
+- **dqm4hep::core::EventStreamer** (*dqm4hep-core* package)
+- **dqm4hep::online::AnalysisModule** (*dqm4hep-online* package)
+- **dqm4hep::online::StandaloneModule** (*dqm4hep-online* package)
+
+# Command line tools
+
+A binary is provided to dump the registered plugins in the console, called *dqm4hep-dump-plugins*. By default, it dumps only the plugins registered in the DQMCore library, that is linked to this binary. You can try to run it simply as :
+
+```shell
+# print only the plugins registered in libDQMCore.so 
+dqm4hep-dump-plugins
+```
+
+or with your own libraries to see what are the compiled plugins :
+
+```shell
+# show our two 'shape' plugins
+export DQM4HEP_PLUGIN_DLL=libShapePlugins.so
+dqm4hep-dump-plugins
+```
+
```

Comment:
Maybe add the output of the command line here

Reply:
Will be added yes